samples/analyze.v1p2beta1.js

```diff
@@ -0,0 +1,348 @@
+
+  /**
+   * TODO(developer): Uncomment the following line before running the sample.
+   */
+  // const gcsUri = 'GCS URI of the video to analyze, e.g. gs://my-bucket/my-video.mp4';
+
+  const request = {
+    inputUri: gcsUri,
+    features: ['OBJECT_TRACKING'],
+    //recommended to use us-east1 for the best latency due to different types of processors used in this region and others
+    locationId: 'us-east1',
+  };
+  // Detects objects in a video
+  const [operation] = await video.annotateVideo(request);
+  //console.log(operation);
+  const results = await operation.promise();
+  console.log('Waiting for operation to complete...');
+  //Gets annotations for video
+  const annotations = results[0].annotationResults[0];
+  const objects = annotations.objectAnnotations;
+  objects.forEach(object => {
+    console.log(`Entity description:  ${object.entity.description}`);
+    console.log(`Entity id: ${object.entity.entityId}`);
+    const time = object.segment;
+    if (time.startTimeOffset.seconds === undefined) {
+      time.startTimeOffset.seconds = 0;
+    }
+    if (time.startTimeOffset.nanos === undefined) {
+      time.startTimeOffset.nanos = 0;
+    }
+    if (time.endTimeOffset.seconds === undefined) {
+      time.endTimeOffset.seconds = 0;
+    }
+    if (time.endTimeOffset.nanos === undefined) {
+      time.endTimeOffset.nanos = 0;
+    }
+    console.log(
+      `Segment: ${time.startTimeOffset.seconds}` +
+        `.${(time.startTimeOffset.nanos / 1e6).toFixed(0)}s to ${
+          time.endTimeOffset.seconds
+        }.` +
+        `${(time.endTimeOffset.nanos / 1e6).toFixed(0)}s`
+    );
+    console.log(`Confidence: ${object.confidence}`);
+    const frame = object.frames[0];
+    const box = frame.normalizedBoundingBox;
+    const timeOffset = frame.timeOffset;
+    if (timeOffset.seconds === undefined) {
+      timeOffset.seconds = 0;
+    }
+    if (timeOffset.nanos === undefined) {
+      timeOffset.nanos = 0;
+    }
+    console.log(
+      `Time offset for the first frame: ${timeOffset.seconds}` +
+        `.${(timeOffset.nanos / 1e6).toFixed(0)}s`
+    );
+    console.log(`Bounding box position:`);
+    console.log(`\tleft   :${box.left}`);
+    console.log(`\ttop    :${box.top}`);
+    console.log(`\tright  :${box.right}`);
+    console.log(`\tbottom :${box.bottom}`);
+  });
+  // [END video_object_tracking_gcs_beta]
+}
+async function analyzeText(path) {
+  //[START video_detect_text_beta]
+  // Imports the Google Cloud Video Intelligence library + Node's fs library
+  const Video = require('@google-cloud/video-intelligence').v1p2beta1;
+  const fs = require('fs');
+  // Creates a client
+  const video = new Video.VideoIntelligenceServiceClient();
+
+  /**
+   * TODO(developer): Uncomment the following line before running the sample.
+   */
+  // const path = 'Local file to analyze, e.g. ./my-file.mp4';
+
+  // Reads a local video file and converts it to base64
+  const file = fs.readFileSync(path);
```

Comment:
I generally don't like using `readFileSync` in examples, because people have a tendency to copy/paste the code from our samples.  Instead of this, can we:

`const file = await util.promisify(fs.readFile)(path)`

Or something like that

Reply:
Fixed in the latest commit